arquillian-chameleon-deployments/arquillian-chameleon-maven-build-deployment/src/test/java/org/arquillian/container/chameleon/deployment/maven/MavenBuildAutomaticDeploymentTest.java

```diff
@@ -2,30 +2,66 @@
 
 import org.jboss.arquillian.test.spi.TestClass;
 import org.jboss.shrinkwrap.api.Archive;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class MavenBuildAutomaticDeploymentTest {
 
+    @Rule
+    public final EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
     @Test
     public void should_get_archive_from_module() {
+        final Archive<?> archive = buildWithTestCaseClass(TestCaseClass.class);
 
-        // given
-        MavenBuildAutomaticDeployment mavenBuildAutomaticDeployment = new MavenBuildAutomaticDeployment();
+        assertThat(archive.toString())
+            .startsWith("arquillian-chameleon-deployment-api");
+    }
 
-        // when
-        final Archive<?> archive = mavenBuildAutomaticDeployment.build(new TestClass(TestCaseClass.class));
+    @Test
+    public void givenTheMavenBuildWithLocalInstallation_whenMavenIsInThePath_thenItMustBuild() {
```

Comment:
I like the idea of elaborating on the context, interaction and expectation for each and every test. That's a really nice improvement. The only concern I have is that if we don't unify across the test suite it can become a bit confusing.

How about meeting mid-way and keep `should_` as test method name, but add the details to every step in the respective comment?

Reply:
Good remark, I agree with your concern.
I'll do the proper change and update the PR.